modules/standard/Socket.md

```diff
@@ -0,0 +1,235 @@
+  ```python
+  proc tcpListener.addr : ipAddr
+  ```
+
+The additional support with `tcpConn` and `tcpListener` will be to enable or disable two of the prominent optimization algorithm for TCP Protocol that are:
+
+- `nagle`
+  This method will set/unset TCP_NODELAY for provided socket.
+  ```python
+  proc nagle(socket:tcpConn, tcpNodeDelay:bool = 1)
+  proc nagle(socket:tcpListener, tcpNodeDelay:bool = 1);
+  proc nagle(socket:udpSocket, tcpNodeDelay:bool = 1);
+  ```
+- `quickAck`
+  This method will set/unset TCP_QUICKACK for provided socket.
+  ```python
+  proc quickAck(socket:tcpConn, tcpQuickAck:bool = 1)
+  proc quickAck(socket:tcpListener, tcpQuickAck:bool = 1);
+  proc quickAck(socket:udpSocket, tcpQuickAck:bool = 1);
+  ```
+
+## UDP Support
+
+Support for UDP protocol will be through a separate constructor for `udpSocket`. They need not be binded or connected as they are connectionless socket.
+```python
+const socketServ:udpSocket = new udpSocket(family=IPFamily.IPv4)
+```
+
+A generic `bind` function will be used which will take in any of the socket three object and bind it to provided `host` , `port`.
+
+**Method**
+
+```
+proc bind(socket, in address:ipAddr, reuseAddr = true)
+```
+
+**Parameters**
+
+```
+socket : tcpListener|tcpConn|udpSocket - socket object to be bound
+address: contains address info about where to bind the socket.
+reuseaddr: Boolean - optional parameter, default value = true
+```
+
+Bind is responsible for binding any socket to a specified address and port. Here the socket can be anything either a UDP or a TCP Socket. listen under the hood will can in bind to bind `tcpListener` to address and port. With either tcpConn or udpSocket that isn't a requirement as the system assigns any free port and localhost address to them by default.
+
+While `tcpConn` is in a connected state when they are created `udpSocket` are connectionless they are just waiting for the user to use them for sending and receiving purpose only. But for sending and receiving purpose user might require a socket to have a known address that is when bind will be useful for `udpSocket`.
+
+## Data transfer on UDP
+
+>UDP I/O can't work the same as TCP because it's stateless there is no persistent connection between the sockets. So because of the absence of a `file` user has to handle I/O by sending in bytes and specify the amount of bytes they are willing to retrieve.
+
+UDP Sockets will have method to call `recvFrom` on it this will return the `addr`, `port` of client socket and `data` being written to the socket.
+
+**Method:**
+```python
+proc socketServ.recvFrom(buffer_len:int, in timeout = new timeval(-1,0)) : (address:ipAddr, data:bytes)
+```
+
+**Parameters**
+
+```
+buffer_len: int - number of bytes to read
+timeout: timeval - defines the time in `(seconds, microseconds)` function should wait till `recvfrom` fails with `TimeoutError`. `(-1, 0)` implies indefinite blocking.
+reuseaddr: Boolean - optional parameter, default value = true
```

Comment:
reuseaddr is not present in the definition above, was adding it here a mistake or should it be present above?

Reply:
shouldn't be here thanks will remove it